src/lookup/lookup.rs

```diff
@@ -0,0 +1,74 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+//
+// Copyright (c) DUSK NETWORK. All rights reserved.
+
+//! Contains the functionality of plookup gates on top of the composer
+
+#![allow(clippy::too_many_arguments)]
+
+use crate::constraint_system::StandardComposer;
+use crate::constraint_system::Variable;
+use dusk_bls12_381::BlsScalar;
+
+impl StandardComposer {
+    /// Adds a plookup gate to the circuit with its corresponding 
+    /// constraints.
+    ///
```

Comment:
would you mind writing the gate equation explicitly here?

Reply:
For now this has been removed @stechu. Only because lookups will be added in different PR's.